### DISCUSSION TITLE
What does "Boost" actually do?

### DISCUSSION THREAD

**Discussion by jeckman** [akirk/friends#363] [Q&A]
Sorry if this is a question for the ActivityPub plugin but the UI appears here in Friends. 

When I "boost" a post in my Friends Feed, what is supposed to actually happen?

I understand from the underlying ActivityPub protocols ( https://seb.jambor.dev/posts/understanding-activitypub/ ) that an "Announce" activity should be sent to followers but not sure what that actually means

If someone is following me on a server like Mastodon, will they see my "boosts" or that dependent on how each server (type or instance?) manages boosts?

**Comment by akirk** [comment]
This is indeed for the Friends plugin, and I agree it could/should be better exposed. I started working on a log in #364 so that you can see in retrospect which ActivityPub actions were sent where.

If you're not afraid of looking at code, take a look here: https://github.com/akirk/friends/blob/e90ae0625d72aded2354833e70a964c3f103f346/feed-parsers/class-feed-parser-activitypub.php#L2091-L2110

We loop over all inboxes of your followers and send them an Announce activity which includes the URL of the post you want to boost.

This will result in their Mastodon feeds to see the boosted post with your name shown as the booster.

**Reply by jeckman** [reply]
Thanks - I'm not seeing those boosts on a Mastodon instance subscribed to my WordPress with ActivityPub running - but that's a bug report for another day when I have time to track the details
